src/benchmark/operators/table_scan_sorted_benchmark.cpp

```diff
@@ -0,0 +1,269 @@
+#include <memory>
+
+#include "../micro_benchmark_basic_fixture.hpp"
+#include "benchmark/benchmark.h"
+#include "expression/expression_functional.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/segment_encoding_utils.hpp"
+#include "storage/table.hpp"
+#include "table_generator.hpp"
+#include "table_scan_sorted_benchmark.hpp"
+#include "type_cast.hpp"
+#include "utils/load_table.hpp"
+
+using namespace opossum::expression_functional;  // NOLINT
+
+namespace opossum {
+
+const int string_size = 512;
+
+void _clear_cache() {
+  std::vector<int> clear = std::vector<int>();
+  clear.resize(500 * 1000 * 1000, 42);
+  for (uint i = 0; i < clear.size(); i++) {
+    clear[i] += 1;
+  }
+  clear.resize(0);
+}
+
+std::shared_ptr<TableColumnDefinitions> create_column_definitions(const DataType data_type) {
+  auto table_column_definitions = std::make_shared<TableColumnDefinitions>();
+
+  // TODO(cmfcmf): Benchmark nullable segments
+  table_column_definitions->emplace_back("a", data_type, false);
+
+  return table_column_definitions;
+}
+
+std::shared_ptr<TableWrapper> create_int_table(const int table_size,
+                                               const EncodingType encoding_type = EncodingType::Unencoded,
+                                               const std::optional<OrderByMode> order_by = std::nullopt) {
+  const auto table_column_definitions = create_column_definitions(DataType::Int);
+
+  std::shared_ptr<Table> table;
+  std::shared_ptr<TableWrapper> table_wrapper;
+
+  table = std::make_shared<Table>(*table_column_definitions, TableType::Data);
+
+  if (!order_by.has_value()) {
+    auto values = std::vector<int>(table_size);
+    for (int i = 0; i < table_size; i++) {
+      values[i] = i;
+    }
+    std::random_shuffle(values.begin(), values.end());
+    for (const auto& num : values) {
+      table->append({num});
```

Comment:
`table->append` is slow, have you considered creating a ValueSegment from `values` and calling `table->append_chunk({segment})`?